syncer/checkpoint_test.go

```diff
@@ -353,6 +361,63 @@ func (s *testCheckpointSuite) testTableCheckPoint(c *C, cp CheckPoint) {
 	newer = cp.IsNewerTablePoint(schema, table, binlog.Location{Position: pos1}, false)
 	c.Assert(newer, IsFalse)
 
+	// save
+	cp.SaveTablePoint(schema, table, binlog.Location{Position: pos2}, nil)
+	newer = cp.IsNewerTablePoint(schema, table, binlog.Location{Position: pos1}, false)
+	c.Assert(newer, IsFalse)
+
+	// delete
+	s.mock.ExpectBegin()
+	s.mock.ExpectExec(deleteCheckPointSQL).WithArgs(cpid, schema, table).WillReturnResult(sqlmock.NewResult(0, 1))
+	s.mock.ExpectCommit()
+	c.Assert(cp.DeleteTablePoint(tctx, schema, table), IsNil)
+	s.mock.ExpectBegin()
+	s.mock.ExpectExec(deleteSchemaPointSQL).WithArgs(cpid, schema).WillReturnResult(sqlmock.NewResult(0, 1))
+	s.mock.ExpectCommit()
+	c.Assert(cp.DeleteSchemaPoint(tctx, schema), IsNil)
+
+	ctx := context.Background()
+
+	// test save with table info and rollback
+	c.Assert(s.tracker.CreateSchemaIfNotExists(schema), IsNil)
+	err = s.tracker.Exec(ctx, schema, "create table "+table+" (c int);")
+	c.Assert(err, IsNil)
+	ti, err := s.tracker.GetTable(schema, table)
+	c.Assert(err, IsNil)
+	cp.SaveTablePoint(schema, table, binlog.Location{Position: pos1}, ti)
+	if rcp, ok := cp.(*RemoteCheckPoint); ok {
```

Comment:
how about forcing cast to `cp.(*RemoteCheckPoint)` without `ok`, because we must have `cp.(*RemoteCheckPoint)` here, otherwise just panic.

if so, ditto for the following lines.

Reply:
changed in https://github.com/pingcap/dm/pull/1049/commits/87b496389bea9361a9595afe304fc1da7c7a582e